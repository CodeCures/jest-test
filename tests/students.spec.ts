import Students from "../src/students";
import User from "../src/user";

describe('Students', () => {

    const user = new User({
        firstName: "Courage",
        lastName: "Francis"
    });

    it('registers a user as student', () => {

        const students = new Students;
        students.addUser(user)

        expect(students.all).toHaveLength(1);
    })

    it('registers more than a user as student', () => {
        
        const user2 = new User({
            firstName: "Foo",
            lastName: "Bar"
        });

        const students = new Students;
        students.addUser(user)
        students.addUser(user2)

        expect(students.all).toHaveLength(2);
    })

    it('matches student records', () => {
        
        const user2 = new User({
            firstName: "Foo",
            lastName: "Bar"
        });

        const students = new Students;
        students.addUser(user)
        students.addUser(user2)

        expect(students.all).toMatchSnapshot();
    })


    it('is an instance of user', () => {

        const students = new Students;
        students.addUser(user)

        expect(students.getUser(0)).toBeInstanceOf(User)
    })

})