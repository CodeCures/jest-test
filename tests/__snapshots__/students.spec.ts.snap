// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Students matches student records 1`] = `
[
  User {
    "firstName": "Courage",
    "lastName": "Francis",
  },
  User {
    "firstName": "Foo",
    "lastName": "Bar",
  },
]
`;
